### PR TITLE
Trigger IdentifyDuplicates when creating client from enrollment form

### PR DESCRIPTION
**Pull request by gigxz**
Also move some MCI-specific things out of Client and into ClientExtension

### CODE REVIEW

File: drivers/hmis/app/graphql/mutations/submit_form.rb [hunk 1]
```diff
@@ -117,10 +117,13 @@ def resolve(input:)
     end
 
     private def perform_side_effects(record)
-      if record.is_a?(Hmis::Hud::Client)
-        if record.new_record?
+      # If a Client record was created or updated, queue up IdentifyDuplicates job. This creates the warehouse destination client.
```
Comment (eanders) [inline drivers/hmis/app/graphql/mutations/submit_form.rb]: Does this get called when someone gets a service as well?  If it does, we should add:
```
enrollment.invalidate_processing!
GrdaWarehouse::Tasks::ServiceHistory::Enrollment.delay.batch_process_unprocessed!
```

Comment (eanders) [reply drivers/hmis/app/graphql/mutations/submit_form.rb]: might need to move the `invalidate_processing!` method into a concern, I think that's just on `GrdaWarehouse::Hud::Enrollment`

**Review by eanders — APPROVED** [review]
Looks good!  If this is also used for providing services, then see the inline suggestion.

File: drivers/hmis/app/graphql/mutations/submit_form.rb (hunk 1 above)
Comment (gigxz) [reply drivers/hmis/app/graphql/mutations/submit_form.rb]: Ok I'm guessing we only need to do that for HUD services, not custom services?

Comment (gigxz) [reply drivers/hmis/app/graphql/mutations/submit_form.rb]: https://github.com/greenriver/hmis-warehouse/pull/3519/commits/f193a95927f363db7defa4f47c8d9c0b0648d33d

File: app/models/concerns/hud_concerns/enrollment.rb [hunk 2]
```diff
@@ -29,5 +29,13 @@ module HudConcerns::Enrollment
     scope :in_project, ->(project_ids) do
       joins(:project).where(p_t[:id].in(project_ids))
     end
+
+    def self.invalidate_processing!
+      update_all(processed_as: nil, processed_hash: nil)
+    end
```
Comment (eanders) [inline app/models/concerns/hud_concerns/enrollment.rb:35]: Oh! I totally didn't realize this was a class method.  That's scary! Anyway, the instance method looks good.  Thanks.

File: drivers/hmis/app/graphql/mutations/submit_form.rb (hunk 1 above)
Comment (eanders) [reply drivers/hmis/app/graphql/mutations/submit_form.rb]: Correct, only HUD services, and CLS when the project is SO

Comment (gigxz) [reply drivers/hmis/app/graphql/mutations/submit_form.rb]: 🧐 okay I'll add for that too

When about when services or cls are deleted, do we need to do this?

Comment (eanders) [reply drivers/hmis/app/graphql/mutations/submit_form.rb]: Yes!  Any change, including changes to the date should trigger this.

Comment (gigxz) [reply drivers/hmis/app/graphql/mutations/submit_form.rb]: Actually, lets move this conversation to https://www.pivotaltracker.com/story/show/186062176. A bit out of scope for this PR and I think there are other places (like ENrollment creation) where this needs to happen

**Pull request merged**